commons/pom.xml

```diff
@@ -14,6 +14,17 @@
    <description>Infinispan Commons</description>
 
    <dependencies>
+      <dependency>
+         <groupId>org.infinispan.protostream</groupId>
+         <artifactId>protostream</artifactId>
+      </dependency>
+
+      <dependency>
+         <groupId>org.infinispan.protostream</groupId>
+         <artifactId>protostream-processor</artifactId>
+         <scope>compile</scope>
```

Comment:
```suggestion
         <scope>provided</scope>
```

Reply:
@danberindei This results in non-commons modules not compiling.

Reply:
Yes, you need to add it to all the modules that use the annotation processor, or you could add it to the parent.

But with `compile` all the applications using `commons` will end up with the `protostream-processor` on their runtime classpath, that's why I moved `component-processor` to the `provided` scope as well:

https://github.com/infinispan/infinispan/commit/8994b1ee6137a6c1a6dde0fd2adc9cf6e50d16d8
https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope